diff --git a/dist/changesets-cli.cjs.js b/dist/changesets-cli.cjs.js
index 577f752a6ba6cb2d8ac7c4740818c7db23fcccfc..6c3388ff2227a4b69d6ea4f4fafa6c6889176d99 100644
--- a/dist/changesets-cli.cjs.js
+++ b/dist/changesets-cli.cjs.js
@@ -714,25 +714,35 @@ async function getPublishTool(cwd) {
   const pm = await packageManagerDetector.detect({
     cwd
   });
-  if (!pm || pm.name !== "pnpm") return {
+  if (!pm) return {
     name: "npm"
   };
-  try {
-    let result = await spawn__default["default"]("pnpm", ["--version"], {
-      cwd
-    });
-    let version = result.stdout.toString().trim();
-    let parsed = semverParse__default["default"](version);
-    return {
-      name: "pnpm",
-      shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
-    };
-  } catch (e) {
+  if (pm.name === "bun") {
     return {
-      name: "pnpm",
-      shouldAddNoGitChecks: false
+      name: "bun"
     };
   }
+  if (pm.name === "pnpm") {
+    try {
+      let result = await spawn__default["default"]("pnpm", ["--version"], {
+        cwd
+      });
+      let version = result.stdout.toString().trim();
+      let parsed = semverParse__default["default"](version);
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
+      };
+    } catch (e) {
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: false
+      };
+    }
+  }
+  return {
+    name: "npm"
+  };
 }
 async function getTokenIsRequired() {
   const {
@@ -831,6 +841,10 @@ async function internalPublish(packageJson, opts, twoFactorState) {
       env: Object.assign({}, process.env, envOverride),
       cwd: opts.cwd,
       stdio: "inherit"
+    }) : publishTool.name === "bun" ? child_process.spawnSync("bun", ["publish", ...publishFlags], {
+      env: Object.assign({}, process.env, envOverride),
+      cwd: opts.publishDir,
+      stdio: "inherit"
     }) : child_process.spawnSync(publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
       env: Object.assign({}, process.env, envOverride),
       stdio: "inherit"
@@ -861,7 +875,10 @@ async function internalPublish(packageJson, opts, twoFactorState) {
   } = publishTool.name === "pnpm" ? await spawn__default["default"]("pnpm", ["publish", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride),
     cwd: opts.cwd
-  }) : await spawn__default["default"](publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
+  }) : publishTool.name === "bun" ? await spawn__default["default"]("bun", ["publish", ...publishFlags], {
+    env: Object.assign({}, process.env, envOverride),
+    cwd: opts.publishDir
+  }) : await spawn__default["default"](publishTool.name, ["publish", opts.publishDir, "--json", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride)
   });
   if (code !== 0) {
diff --git a/dist/changesets-cli.esm.js b/dist/changesets-cli.esm.js
index 1c568a79190bf5fa05ed75799570b9c8eb902cdb..a9150dccac4124cf689673ba7ff949a0974bb40a 100644
--- a/dist/changesets-cli.esm.js
+++ b/dist/changesets-cli.esm.js
@@ -677,25 +677,35 @@ async function getPublishTool(cwd) {
   const pm = await detect({
     cwd
   });
-  if (!pm || pm.name !== "pnpm") return {
+  if (!pm) return {
     name: "npm"
   };
-  try {
-    let result = await spawn$1("pnpm", ["--version"], {
-      cwd
-    });
-    let version = result.stdout.toString().trim();
-    let parsed = semverParse(version);
-    return {
-      name: "pnpm",
-      shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
-    };
-  } catch (e) {
+  if (pm.name === "bun") {
     return {
-      name: "pnpm",
-      shouldAddNoGitChecks: false
+      name: "bun"
     };
   }
+  if (pm.name === "pnpm") {
+    try {
+      let result = await spawn$1("pnpm", ["--version"], {
+        cwd
+      });
+      let version = result.stdout.toString().trim();
+      let parsed = semverParse(version);
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: (parsed === null || parsed === void 0 ? void 0 : parsed.major) === undefined ? false : parsed.major >= 5
+      };
+    } catch (e) {
+      return {
+        name: "pnpm",
+        shouldAddNoGitChecks: false
+      };
+    }
+  }
+  return {
+    name: "npm"
+  };
 }
 async function getTokenIsRequired() {
   const {
@@ -794,6 +804,10 @@ async function internalPublish(packageJson, opts, twoFactorState) {
       env: Object.assign({}, process.env, envOverride),
       cwd: opts.cwd,
       stdio: "inherit"
+    }) : publishTool.name === "bun" ? spawnSync("bun", ["publish", ...publishFlags], {
+      env: Object.assign({}, process.env, envOverride),
+      cwd: opts.publishDir,
+      stdio: "inherit"
     }) : spawnSync(publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
       env: Object.assign({}, process.env, envOverride),
       stdio: "inherit"
@@ -824,7 +838,10 @@ async function internalPublish(packageJson, opts, twoFactorState) {
   } = publishTool.name === "pnpm" ? await spawn$1("pnpm", ["publish", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride),
     cwd: opts.cwd
-  }) : await spawn$1(publishTool.name, ["publish", opts.publishDir, ...publishFlags], {
+  }) : publishTool.name === "bun" ? await spawn$1("bun", ["publish", ...publishFlags], {
+    env: Object.assign({}, process.env, envOverride),
+    cwd: opts.publishDir
+  }) : await spawn$1(publishTool.name, ["publish", opts.publishDir, "--json", ...publishFlags], {
     env: Object.assign({}, process.env, envOverride)
   });
   if (code !== 0) {
